iwyu

petersteneteg opened the pull request:
Thank you for contributing to the Inviwo repository! Great code makes great apps so please ensure the following:

**Always**
- [ ] Code is error and warning free
- [ ] Code follows the [Inviwo coding guidelines](https://github.com/inviwo/inviwo/wiki/Coding-Conventions)

**New features**
- [ ] Code / Processors follows the [Inviwo documentation guidelines](https://github.com/inviwo/inviwo/wiki/Documentation-guide)
- [ ] Code / processors are tested in [unittest and/or regresion tests](https://github.com/inviwo/inviwo/wiki/Manual#testing)

**Updated feature / Fixes** 
- [ ] Relevant documentation has been updated

**What evineroment has the code been compiled and tested on?** 
- Operating system
- Compiler/IDE
- QT/CMake/Python versions

Are you are unsure how to address these points? Please make a note here so that we can help out:

ResearchDaniel reviewed (dismissed):
No idea how to use it but the results look ok.
Could you add a ReadMe.md file in the folder where you explain that it is IWYU stands for include what you use and how to use it?

JochenJankowai commented on tools/iwyu/glmimp.py:
```diff
@@ -0,0 +1,57 @@
+    return alist[:len(prefix)] == prefix
+
+
+glm_base = "/Users/peter/Work/Inviwo/inviwo/ext/glm/glm"
```
Is this a generated file? Otherwise, this path should probably be relative.

petersteneteg replied:
Yes, but you only have to do it ones kind of, or of glm does some large changes...

JochenJankowai commented:
I am not too familiar with Jenkins' capabilities but isn't this something that could run there?

petersteneteg commented:
Yes, I would like to run it on Jenkins or GitHub Actions at some point... But at the moment it changes quite a lot of stuff so the interaction of it has to be done with a bit of care to avoid breaking way to much code...

martinfalk commented on modules/basegl/src/processors/multichannelraycaster.cpp:
```diff
@@ -32,9 +32,11 @@
 #include <modules/opengl/volume/volumegl.h>
 #include <modules/opengl/image/layergl.h>
 #include <modules/opengl/texture/textureunit.h>
+
```
empty lines?

martinfalk reviewed (dismissed):
what about qt6_x header files?

petersteneteg committed:
Plotting: iwyu

petersteneteg committed:
PlottingGL: iwyu

petersteneteg committed:
UserInterfaceGL: iwyu

petersteneteg committed:
MeshRenderingGL: iwyu

petersteneteg committed:
PostProcessing: iwyu

petersteneteg committed:
VectorFieldVisualizationGL: iwyu

# Conflicts:
#	modules/vectorfieldvisualizationgl/src/processors/3d/streamparticles.cpp

petersteneteg committed:
Volume: iwyu

petersteneteg committed:
Webbrowser: iwyu

# Conflicts:
#	modules/webbrowser/src/processors/processorcefsynchronizer.cpp
#	modules/webbrowser/src/properties/propertycefsynchronizer.cpp

petersteneteg committed:
DataFrame: iwyu

# Conflicts:
#	modules/dataframe/include/inviwo/dataframe/datastructures/dataframe.h

petersteneteg committed:
VectorFieldVisualization: iwyu

petersteneteg committed:
Nifti: iwyu

petersteneteg committed:
FontRendering: iwyu

petersteneteg committed:
GLFW: iwyu

petersteneteg committed:
JSON: iwyu

petersteneteg committed:
Eigen: iwyu

petersteneteg committed:
BrushingAndLinking: iwyu

petersteneteg committed:
Base: iwyu

petersteneteg committed:
BaseGL: incfix

petersteneteg committed:
PlottingGL incfix

petersteneteg committed:
PostProcessing: incfix

petersteneteg committed:
Python3GL: incfix

petersteneteg committed:
UserinterfaceGL: incfix

petersteneteg committed:
OpenGL: iwyu

petersteneteg committed:
BaseGL: iwyu

petersteneteg committed:
Python3: iwyu

petersteneteg committed:
PVM: iwyu

petersteneteg committed:
PNG: iwyu

petersteneteg committed:
CIMG: iwyu

petersteneteg committed:
Assimp: iwyu

petersteneteg committed:
AnimationQt: incfix

petersteneteg committed:
Animation: iwyu

petersteneteg committed:
Python3: incfix

petersteneteg committed:
UserinterfaceGL: incfix

petersteneteg committed:
DataFrame: Incfix

petersteneteg committed:
GLFW: incfix

petersteneteg committed:
CMake: define _USE_MATH_DEFINES on windows, so including cmath also defiines M_PI etc

petersteneteg committed:
DataFrameQt: incfix

petersteneteg committed:
OpenGLQt: incfix

petersteneteg committed:
Python3Qt: incfix

petersteneteg committed:
QtEditor: incfix

petersteneteg committed:
QtWidgets: iwyu

petersteneteg committed:
QtWidgets: remove qt blanket warning ignores

petersteneteg committed:
Tools: iwyu Qt update

petersteneteg committed:
QtWidget: iwyu update

petersteneteg committed:
Tools: iwyu qt update
